### PR TITLE
Speculative execution stats fixes

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java [hunk 1]
```diff
@@ -2369,14 +2370,10 @@ private void recordFinishStats()
         {
             long finishTime = System.currentTimeMillis();
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java:2371]: This is wrong. currentTimeMillis is using a system clock that can go backwards. To measure latencies nanoTime should be used (monotonically increasing)

Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java:2371]: Does not matter for this use - but I will change

Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java:2371]: Sure. It was a general comment. I've seen some user reported errors that duration was negative

**Comment by losipiuk** [comment]
ac

**Pull request merged**